### PR TITLE
8338748: [17u,21u] Test Disconnect.java compile error: cannot find symbol after JDK-8299813

### PR DESCRIPTION
**Pull request by sendaoYan**
Hi all,
Test `java/nio/channels/DatagramChannel/Disconnect.java` compile error: `cannot find symbol InetAddress.ofLiteral` after JDK-8299813. There is no `InetAddress.ofLiteral` API in jdk21u and jdk17u, the `InetAddress.ofLiteral` API was added by [JDK-8272215](https://bugs.openjdk.org/browse/JDK-8272215) in jdk22.
Change `InetAddress.ofLiteral`to `InetAddress.getByName`, change has been verified, test fix only, no risk.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] [JDK-8338748](https://bugs.openjdk.org/browse/JDK-8338748) needs maintainer approval
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8338748](https://bugs.openjdk.org/browse/JDK-8338748): [17u,21u] Test Disconnect.java compile error: cannot find symbol after JDK-8299813 (**Bug** - P3 - Approved)

### Reviewers
 * [Severin Gehwolf](https://openjdk.org/census#sgehwolf) (@jerboaa - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk21u-dev.git pull/939/head:pull/939` \
`$ git checkout pull/939`

Update a local copy of the PR: \
`$ git checkout pull/939` \
`$ git pull https://git.openjdk.org/jdk21u-dev.git pull/939/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 939`

View PR using the GUI difftool: \
`$ git pr show -t 939`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk21u-dev/pull/939.diff">https://git.openjdk.org/jdk21u-dev/pull/939.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk21u-dev/pull/939#issuecomment-2302427977)

### CODE REVIEW

**Comment by sendaoYan** [comment]
/approval request Fix test bug, the change has been verified, test fix only, no risk.

**Comment by shipilev** [comment]
Is this a 17u/21u-specific fix? I believe the style is to prepend the synopsis with `[17u, 21u] ...` then.

Missing context: [JDK-8272215](https://bugs.openjdk.org/browse/JDK-8272215) is only available in JDK 22, this is why this fails.

**Comment by sendaoYan** [comment]
> Is this a 17u/21u-specific fix? I believe the style is to prepend the synopsis with `[17u, 21u] ...` then.
> 
> Missing context: [JDK-8272215](https://bugs.openjdk.org/browse/JDK-8272215) is only available in JDK 22, this is why this fails.

Thanks for the reminding, it's a 17u/21u-specific fix, the jbs title has been updated, and the missing context has been added.

**Comment by sendaoYan** [comment]
Before this PR has been reviewed, the approval label should not seted.
/approval cancel

File: test/jdk/java/nio/channels/DatagramChannel/Disconnect.java
Comment (jerboaa) [inline test/jdk/java/nio/channels/DatagramChannel/Disconnect.java:55]: `Inet4Address` doesn't override `getByName()` we should use `InetAddress` instead.

```suggestion
                InetAddress lo4 = InetAddress.getByName("127.0.0.1");
```

File: test/jdk/java/nio/channels/DatagramChannel/Disconnect.java
Comment (jerboaa) [inline test/jdk/java/nio/channels/DatagramChannel/Disconnect.java:65]:
```suggestion
                InetAddress lo6 = InetAddress.getByName("::1");
```

**Review by jerboaa — COMMENTED** [review]
The point here is, AFAIK, to do parsing without the reverse look-up. The [javadoc](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/net/InetAddress.html#getByName(java.lang.String)) states for `getByName()`:

```
The host name can either be a machine name, such as "www.example.com", or a textual representation of its IP address. If a literal IP address is supplied, only the validity of the address format is checked. 
```

So the replacement seems OK in principle. Did you run the test?

**Comment by sendaoYan** [comment]
> So the replacement seems OK in principle. Did you run the test?

Yes, the change has been verified locally.

File: test/jdk/java/nio/channels/DatagramChannel/Disconnect.java
Comment (sendaoYan) [reply test/jdk/java/nio/channels/DatagramChannel/Disconnect.java]: Same to above.

**Review by jerboaa — APPROVED** [review]

**Comment by sendaoYan** [comment]
Thanks for the review.

/approval request Fix test bug, the change has been verified locally, test fix only, no risk.

**Comment by GoeLin** [comment]
GHA failure is the well known ClhsdbFindPC issue. Unrelated.

**Comment by sendaoYan** [comment]
The GHA test [runner](https://github.com/sendaoYan/jdk21u-dev-ysd/actions/runs/10774948680/job/29907380098) report two failures:

1. ClhsdbDumpheap.java fails, which has been recorded by [JDK-8247940](https://bugs.openjdk.org/browse/JDK-8247940), it's unreleated to this PR.
2. ClhsdbFindPC.java#no-xcomp-process fails, which has been recorded by [JDK-8288429](https://bugs.openjdk.org/browse/JDK-8288429), it's unreleated to this PR.

**Comment by sendaoYan** [comment]
Thanks all for the review and approved.
/integrate

**Comment by sendaoYan** [comment]
Thanks for the sponsor.

**Pull request closed without merging**
